agent/eni/networkutils/utils_windows.go

```diff
@@ -201,3 +259,37 @@ func getAdapterAddresses() ([]*windows.IpAdapterAddresses, error) {
 	}
 	return aas, nil
 }
+
+// getIfTable2Ex invokes GetIfTable2Ex Win32 API to retrieve the interface table.
+func getIfTable2Ex() (*mibIfTable2, error) {
+	var tab *mibIfTable2
+
+	// getIfTable2Ex function retrieves the MIB-II interface table.
+	// https://docs.microsoft.com/en-us/windows/desktop/api/netioapi/nf-netioapi-getiftable2ex
+	retVal, _, _ := procGetIfTable2Ex.Call(uintptr(0), uintptr(unsafe.Pointer(&tab)))
+	if retVal != 0 {
+		return nil, errors.Errorf("error occured while calling GetIfTable2Ex: %s", syscall.Errno(retVal))
+	}
+
+	return tab, nil
+}
+
+// getIfEntry2Ex invokes GetIfEntry2Ex Win32 API to retrieve the MIB_IF_ROW2 for the interface with given LUID.
+func getIfEntry2Ex(ifRow *MibIfRow2) (*MibIfRow2, error) {
```

Comment:
Did we add this extra function just for unit testing?

Reply:
Yes we did :)